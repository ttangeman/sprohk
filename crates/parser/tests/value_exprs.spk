var a = 1 * 2 + 3;
var b = 1 + 2 * 3;
var c = -1 + 2;
var d = -(1 + 2);
var e = 2 * (4 + 4);