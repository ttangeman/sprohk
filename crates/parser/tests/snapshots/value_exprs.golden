node 0 {
  kind: ValueExpr
  tokens: [3..4]
  data: {
    literal: '1'
  }
}
node 1 {
  kind: ValueExpr
  tokens: [5..6]
  data: {
    literal: '2'
  }
}
node 2 {
  kind: ValueExpr
  tokens: [3..6]
  data: {
    op: '*'
    lhs: 0
    rhs: 1
  }
}
node 3 {
  kind: ValueExpr
  tokens: [7..8]
  data: {
    literal: '3'
  }
}
node 4 {
  kind: ValueExpr
  tokens: [3..8]
  data: {
    op: '+'
    lhs: 2
    rhs: 3
  }
}
node 5 {
  kind: VarDecl
  tokens: [0..9]
  data: {
    specifier: Var
    name: 'a'
    assign_expr: 4
  }
}
node 6 {
  kind: ValueExpr
  tokens: [12..13]
  data: {
    literal: '1'
  }
}
node 7 {
  kind: ValueExpr
  tokens: [14..15]
  data: {
    literal: '2'
  }
}
node 8 {
  kind: ValueExpr
  tokens: [16..17]
  data: {
    literal: '3'
  }
}
node 9 {
  kind: ValueExpr
  tokens: [14..17]
  data: {
    op: '*'
    lhs: 7
    rhs: 8
  }
}
node 10 {
  kind: ValueExpr
  tokens: [12..17]
  data: {
    op: '+'
    lhs: 6
    rhs: 9
  }
}
node 11 {
  kind: VarDecl
  tokens: [9..18]
  data: {
    specifier: Var
    name: 'b'
    assign_expr: 10
  }
}
node 12 {
  kind: ValueExpr
  tokens: [22..23]
  data: {
    literal: '1'
  }
}
node 13 {
  kind: ValueExpr
  tokens: [21..23]
  data: {
    op: '-'
    rhs: 12
  }
}
node 14 {
  kind: ValueExpr
  tokens: [24..25]
  data: {
    literal: '2'
  }
}
node 15 {
  kind: ValueExpr
  tokens: [21..25]
  data: {
    op: '+'
    lhs: 13
    rhs: 14
  }
}
node 16 {
  kind: VarDecl
  tokens: [18..26]
  data: {
    specifier: Var
    name: 'c'
    assign_expr: 15
  }
}
node 17 {
  kind: ValueExpr
  tokens: [31..32]
  data: {
    literal: '1'
  }
}
node 18 {
  kind: ValueExpr
  tokens: [33..34]
  data: {
    literal: '2'
  }
}
node 19 {
  kind: ValueExpr
  tokens: [31..34]
  data: {
    op: '+'
    lhs: 17
    rhs: 18
  }
}
node 20 {
  kind: ValueExpr
  tokens: [29..35]
  data: {
    op: '-'
    rhs: 19
  }
}
node 21 {
  kind: VarDecl
  tokens: [26..36]
  data: {
    specifier: Var
    name: 'd'
    assign_expr: 20
  }
}
node 22 {
  kind: ValueExpr
  tokens: [39..40]
  data: {
    literal: '2'
  }
}
node 23 {
  kind: ValueExpr
  tokens: [42..43]
  data: {
    literal: '4'
  }
}
node 24 {
  kind: ValueExpr
  tokens: [44..45]
  data: {
    literal: '4'
  }
}
node 25 {
  kind: ValueExpr
  tokens: [42..45]
  data: {
    op: '+'
    lhs: 23
    rhs: 24
  }
}
node 26 {
  kind: ValueExpr
  tokens: [39..46]
  data: {
    op: '*'
    lhs: 22
    rhs: 25
  }
}
node 27 {
  kind: VarDecl
  tokens: [36..47]
  data: {
    specifier: Var
    name: 'e'
    assign_expr: 26
  }
}
